### DISCUSSION TITLE
GTK and async data

### DISCUSSION THREAD

**Discussion by knuxbbs** [GtkSharp/GtkSharp#310] [Q&A] [answered]
How to get a label's text from an async method before show a window? There is any guidance about this?

Because it's come from an async method, I don't want to do this in the window constructor.

**Comment by harry-cpp** [comment, marked as answer]
OnShow event with Glib.ConnectBefore?

```cs
window.Shown += OnShown;
```

```cs
[GLib.ConnectBefore]
private async void OnShown(object sender, EventArgs args)
{

}
```

**Reply by knuxbbs** [reply]
Tested here. Thanks.
